integration/tests/epochs/recover_epoch/recover_epoch_efm_test.go

```diff
@@ -19,16 +23,79 @@ type RecoverEpochSuite struct {
+// 4. Ensure expected EpochRecover event is emitted.
+// Currently, this test does not test the processing of the EpochRecover event see this issue: https://github.com/onflow/flow-go/issues/6164
 func (s *RecoverEpochSuite) TestRecoverEpoch() {
-	s.AwaitEpochPhase(context.Background(), 0, flow.EpochPhaseSetup, 20*time.Second, time.Second)
-	fmt.Println("in epoch phase setup")
+	// 1. Manually trigger EFM
+	// wait until the epoch setup phase to force network into EFM
+	s.AwaitEpochPhase(s.Ctx, 0, flow.EpochPhaseSetup, 10*time.Second, 500*time.Millisecond)
+	// pausing consensus node will force the network into EFM
+	ln := s.GetContainersByRole(flow.RoleCollection)[0]
+	_ = ln.Pause()
+	s.AwaitFinalizedView(s.Ctx, 32, 2*time.Minute, 500*time.Millisecond)
+	// start the paused execution node now that we are in EFM
+	require.NoError(s.T(), ln.Start())
 
-	sns := s.GetContainersByRole(flow.RoleConsensus)
-	_ = sns[0].Pause()
+	// get the latest snapshot and start new container with it
+	epoch1FinalView, err := s.Net.BootstrapSnapshot.Epochs().Current().FinalView()
+	require.NoError(s.T(), err)
 
-	// @TODO: trigger EFM manually
+	// wait for at least the first block of the next epoch to be sealed so that we can
+	// ensure that we are still in the same epoch after the final view of that epoch indicating we are in EFM
+	s.TimedLogf("waiting for epoch transition (finalized view %d) before pausing container", epoch1FinalView+1)
+	s.AwaitFinalizedView(s.Ctx, epoch1FinalView+1, 2*time.Minute, 500*time.Millisecond)
+	s.TimedLogf("observed finalized view %d -> pausing container", epoch1FinalView+1)
+
+	//assert transition to second epoch did not happen
+	//if counter is still 0, epoch emergency fallback was triggered as expected
+	s.AssertInEpoch(s.Ctx, 0)
+
+	// 2. Generate transaction arguments for epoch recover transaction.
+	// generate epoch recover transaction args
+	collectionClusters := uint64(1)
+	numViewsInRecoveryEpoch := uint64(80)
+	numViewsInStakingAuction := uint64(2)
+	epochCounter := uint64(1)
+	targetDuration := uint64(3000)
+	targetEndTime := uint64(4000)
+	out := fmt.Sprintf("%s/recover-epoch-tx-args.json", s.Net.BootstrapDir)
+	s.executeEFMRecoverTXArgsCMD(
+		collectionClusters,
+		numViewsInRecoveryEpoch,
+		numViewsInStakingAuction,
+		epochCounter,
+		targetDuration,
+		targetEndTime,
+		out,
+	)
+	b, err := os.ReadFile(out)
+	require.NoError(s.T(), err)
+
+	// 3. Submit recover epoch transaction to the network.
+	// submit the recover epoch transaction
+	txArgs, err := utils.ParseJSON(b)
+	require.NoError(s.T(), err)
+	env := utils.LocalnetEnv()
+	result := s.recoverEpoch(env, txArgs)
+	require.NoError(s.T(), result.Error)
+	require.Equal(s.T(), result.Status, sdk.TransactionStatusSealed)
+
+	// 3. Ensure EpochRecover event was emitted.
+	eventType := ""
+	for _, evt := range result.Events {
+		if strings.Contains(evt.Type, "FlowEpoch.EpochRecover") {
+			eventType = evt.Type
+			break
+		}
+	}
```

Comment:
need to ensure that `eventType` was found in loop above, otherwise we might be comparing with empty string.

Reply:
https://github.com/onflow/flow-go/pull/6156/commits/582ca2468de44b6211f050f526835e57e94525af